### PR TITLE
Added examples and docs for nft mint, send, burn

### LINKED ISSUE #415: Create NFTs examples with related docs
## Description

Create examples for the `NFTs` section. It should contain:

- [ ] Burn
- [ ] Transfer
- [ ] Transfer with Storage Deposit Return and Expiration
- [ ] Claim
- [ ] Claim under conditional Transfer
- [ ] Introduction
- [ ] Mint
- [ ] Create NFT collections

## Are you planning to do it yourself in a pull request?

Yes, someone from out team will work on it.

### PR DESCRIPTION
**Pull request by anistark**
# Description of change

Added docs for nft mint, send, burn. 
Also renamed a file with spell fix.

## Links to any relevant issues

Part of issue #415 

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Documentation Fix

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: bindings/nodejs/examples/how_tos/nfts/burn_nft.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    Wallet
+} from '@iota/sdk';
+
+// In this example we will burn an existing nft output.
+//
+// Make sure that `example.stronghold` and `example.walletdb` already exist by
+// running the `how_tos/accounts-and-addresses/create-wallet` example!
+//
+// Rename `.env.example` to `.env` first, then run
+// yarn run-example ./wallet/14-burn-nft.ts
```
Comment (Thoralf-M) [inline bindings/nodejs/examples/how_tos/nfts/burn_nft.ts]:
```suggestion
// yarn run-example ./how_tos/nfts/burn_nft.ts
```

File: bindings/nodejs/examples/how_tos/nfts/burn_nft.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            throw new Error(
+                `No NFT available in account '${process.env.ACCOUNT_ALIAS_1}'`,
+            );
+        }
+        // Get the first nft
+        const nftId = balance.nfts[0];
+
+        console.log(`Balance BEFORE burning:\n`, balance);
+
+        // Burn a native token
```
Comment (Thoralf-M) [inline bindings/nodejs/examples/how_tos/nfts/burn_nft.ts]:
```suggestion
        // Burn an NFT
```

File: bindings/nodejs/examples/how_tos/nfts/send_nft.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { SendNftParams, Wallet } from '@iota/sdk';
+
+// The address to send the tokens to
```
Comment (Thoralf-M) [inline bindings/nodejs/examples/how_tos/nfts/send_nft.ts]:
```suggestion
// The address to send the NFT to
```

File: bindings/nodejs/examples/how_tos/nfts/send_nft.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+const RECV_ADDRESS =
+    'rms1qpszqzadsym6wpppd6z037dvlejmjuke7s24hm95s9fg9vpua7vluaw60xu';
+
+// In this example we will send an NFT (Non-fungible token).
+//
+// Make sure that `example.stronghold` and `example.walletdb` already exist by
+// running the `how_tos/accounts-and-addresses/create-wallet` example!
+//
+// Rename `.env.example` to `.env` first, then run
+// yarn run-example ./wallet/08-send-nft.ts
```
Comment (Thoralf-M) [inline bindings/nodejs/examples/how_tos/nfts/send_nft.ts]:
```suggestion
// yarn run-example ./how_tos/nfts/send_nft.ts
```

File: bindings/nodejs/examples/how_tos/nfts/mint_nft.ts [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    AddressUnlockCondition,
+    Ed25519Address,
+    IssuerFeature,
+    MintNftParams,
+    SenderFeature,
+    utf8ToHex,
+    Utils,
+    Wallet
+} from '@iota/sdk';
+
```
Comment (Thoralf-M) [inline bindings/nodejs/examples/how_tos/nfts/mint_nft.ts:14]: Missing example description and run command

File: bindings/nodejs/examples/wallet/10-mint-nft.ts
Comment (Thoralf-M) [inline bindings/nodejs/examples/wallet/10-mint-nft.ts:1]: No need to keep this file if we have how_tos/nfts/mint_nft.ts, same for the other examples, if we have them in the how-tos folder, we can delete the other one

File: bindings/python/examples/how_tos/nfts/mint_nft.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+# Sync account with the node
+response = account.sync()
+
+outputs = [{
+    "immutableMetadata": utf8_to_hex("some immutable nft metadata"),
+}]
+
+transaction = account.prepare_mint_nfts(outputs).send()
+
```
Comment (Thoralf-M) [inline bindings/python/examples/how_tos/nfts/mint_nft.py]:
```suggestion
```

File: bindings/python/examples/how_tos/nfts/send_nft.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+# Sync account with the node
+response = account.sync()
+
+outputs = [{
+    "address": "rms1qpszqzadsym6wpppd6z037dvlejmjuke7s24hm95s9fg9vpua7vluaw60xu",
+    "nftId": "0x17f97185f80fa56eab974de6b7bbb80fa812d4e8e37090d166a0a41da129cebc",
+}]
+
+transaction = account.prepare_send_nft(outputs).send()
+
```
Comment (Thoralf-M) [inline bindings/python/examples/how_tos/nfts/send_nft.py]:
```suggestion
```

File: documentation/sdk/docs/how_tos/nfts/01_mint_nft.mdx
Comment (Thoralf-M) [inline documentation/sdk/docs/how_tos/nfts/01_mint_nft.mdx:1]: Do we really want the number prefix here?

File: documentation/sdk/docs/how_tos/nfts/01_mint_nft.mdx [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+This How-To will show you how to mint nft.
+
+:::tip Sync Options
+
+If you want to list incoming transactions you need to specify that in your `sync` call. 
+Transactions for outputs that you synced before with other sync options, will not be requested later.
+Also only transactions that weren't pruned on the point of syncing will be listed.
+
+:::tip
```
Comment (Thoralf-M) [inline documentation/sdk/docs/how_tos/nfts/01_mint_nft.mdx]: Not interesting here
```suggestion
```

File: documentation/sdk/docs/how_tos/nfts/02_send_nft.mdx [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+This How-To will show you how to mint nft.
+
+:::tip Sync Options
+
+If you want to list incoming transactions you need to specify that in your `sync` call. 
+Transactions for outputs that you synced before with other sync options, will not be requested later.
+Also only transactions that weren't pruned on the point of syncing will be listed.
+
+:::tip
```
Comment (Thoralf-M) [inline documentation/sdk/docs/how_tos/nfts/02_send_nft.mdx]: Not interesting here
```suggestion
```

File: documentation/sdk/docs/how_tos/nfts/03_burn_nft.mdx [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+
+This How-To will show you how to mint nft.
+
+:::tip Sync Options
+
+If you want to list incoming transactions you need to specify that in your `sync` call. 
+Transactions for outputs that you synced before with other sync options, will not be requested later.
+Also only transactions that weren't pruned on the point of syncing will be listed.
+
+:::tip
```
Comment (Thoralf-M) [inline documentation/sdk/docs/how_tos/nfts/03_burn_nft.mdx]: Not interesting here
```suggestion
```

File: documentation/sdk/docs/how_tos/nfts/03_burn_nft.mdx [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+---
+
+import CodeBlock from '@theme/CodeBlock';
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+import RustCode from "!!raw-loader!../../../../../sdk/examples/how_tos/nfts/burn_nft.rs";
+import NodejsCode from "!!raw-loader!../../../../../bindings/nodejs/examples/how_tos/nfts/burn_nft.ts";
+import PythonCode from "!!raw-loader!../../../../../bindings/python/examples/how_tos/nfts/burn_nft.py";
+
+This How-To will show you how to mint nft.
```
Comment (Thoralf-M) [inline documentation/sdk/docs/how_tos/nfts/03_burn_nft.mdx]:
```suggestion
This How-To will show you how to burn an NFT.
```

File: sdk/examples/how_tos/nfts/burn_nft.rs [hunk 12]
```diff
@@ -0,0 +1,52 @@
+// Copyright 2022 IOTA Stiftung
```
Comment (Thoralf-M) [inline sdk/examples/how_tos/nfts/burn_nft.rs]:
```suggestion
// Copyright 2023 IOTA Stiftung
```

File: sdk/examples/how_tos/nfts/mint_nft.rs [hunk 13]
```diff
@@ -0,0 +1,96 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! In this example we will mint a native token.
```
Comment (Thoralf-M) [inline sdk/examples/how_tos/nfts/mint_nft.rs]:
```suggestion
//! In this example we will mint an NFT.
```

File: sdk/examples/how_tos/nfts/mint_nft.rs [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+            "rms1qpszqzadsym6wpppd6z037dvlejmjuke7s24hm95s9fg9vpua7vluaw60xu",
+        )?),
+        sender: Some(Bech32Address::try_from_str(
+            "rms1qpllaj0pyveqfkwxmnngz2c488hfdtmfrj3wfkgxtk4gtyrax0jaxzt70zy",
+        )?),
+        metadata: Some(b"some NFT metadata".to_vec()),
+        tag: Some(b"some NFT tag".to_vec()),
+        issuer: Some(Bech32Address::try_from_str(
+            "rms1qpllaj0pyveqfkwxmnngz2c488hfdtmfrj3wfkgxtk4gtyrax0jaxzt70zy",
+        )?),
```
Comment (Thoralf-M) [inline sdk/examples/how_tos/nfts/mint_nft.rs]: Sender and issuer will not work with a different mnemonic, should probably also just use the first account address as in the nodejs examples

File: sdk/examples/how_tos/nfts/send_nft.rs [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+        // Wait for transaction to get included
+        let block_id = account
+            .retry_transaction_until_included(&transaction.transaction_id, None, None)
+            .await?;
+
+        println!(
+            "Block included: {}/block/{}",
+            std::env::var("EXPLORER_URL").unwrap(),
+            block_id
+        );
```
Comment (Thoralf-M) [inline sdk/examples/how_tos/nfts/send_nft.rs]: Different output than in the `## Expected Output`

**Comment by Dr-Electron** [comment]
Format check for nodejs bindings is failing :)

File: documentation/sdk/docs/how_tos/nfts/01_mint_nft.mdx
Comment (anistark) [reply documentation/sdk/docs/how_tos/nfts/01_mint_nft.mdx:1]: On the sidebar, burn was coming above mint. Hence, added.

File: sdk/examples/how_tos/nfts/send_nft.rs (hunk 15 above)
Comment (anistark) [reply sdk/examples/how_tos/nfts/send_nft.rs]: Yeah. Noted the output for nodejs. 
Need to work on this. Either make them all same output or add diff outputs for each in tabs.

File: .gitignore [hunk 16]
```diff
@@ -3,6 +3,7 @@
 **/target
 *.pyc
 **/*.rs.bk
+yarn.lock
```
Comment (thibault-martinez) [inline .gitignore]: We sure about this? I don't know js enough but it wasn't done so far, any reason why? If we really ignore it, we should also remove the ones that have been pushed, apparently 4

File: bindings/nodejs/examples/how_tos/nfts/burn_nft.ts [hunk 17]
```diff
@@ -0,0 +1,70 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+    Wallet
+} from '@iota/sdk';
```
Comment (thibault-martinez) [inline bindings/nodejs/examples/how_tos/nfts/burn_nft.ts]: This doesn't look formated

File: documentation/sdk/docs/how_tos/nfts/01_mint_nft.mdx
Comment (Dr-Electron) [reply documentation/sdk/docs/how_tos/nfts/01_mint_nft.mdx:1]: I think we prefer sorting in the sidebar. It's way easier to move files around that way. So I would change it

File: sdk/examples/how_tos/nfts/send_nft.rs (hunk 15 above)
Comment (Dr-Electron) [reply sdk/examples/how_tos/nfts/send_nft.rs]: Until now we tried to have the same output in all examples. And I think that's quite nice, also as some sort of check

File: documentation/sdk/docs/how_tos/nfts/01_mint_nft.mdx
Comment (thibault-martinez) [reply documentation/sdk/docs/how_tos/nfts/01_mint_nft.mdx:1]: Yeah let's remove the numbers if we can, it's been annoying everyone since forever

Comment (anistark) [reply documentation/sdk/docs/how_tos/nfts/01_mint_nft.mdx:1]: ok. removed.

File: .gitignore (hunk 16 above)
Comment (anistark) [reply .gitignore]: On project level, it's fine. Inside docs, while testing one extra one got created. So, removed it.

File: bindings/nodejs/examples/how_tos/nfts/mint_nft.ts [hunk 18]
```diff
@@ -0,0 +1,129 @@
...
+        // Build an NFT manually by using the `NftOutputBuilder`
+        const client = await wallet.getClient();
+
+        const hexAddress = Utils.bech32ToHex(senderAddress);
+        const output = await client.buildNftOutput({
+            amount: NFT2_AMOUNT,
+            nftId: '0x0000000000000000000000000000000000000000000000000000000000000000',
+            unlockConditions: [
+                new AddressUnlockCondition(new Ed25519Address(hexAddress)),
```
Comment (Thoralf-M) [inline bindings/nodejs/examples/how_tos/nfts/mint_nft.ts]: This should be `Utils.bech32ToHex(NFT1_OWNER_ADDRESS)`?

Comment (anistark) [reply bindings/nodejs/examples/how_tos/nfts/mint_nft.ts]: Yes. Changing it.

Comment (Thoralf-M) [reply bindings/nodejs/examples/how_tos/nfts/mint_nft.ts]: Only in the AddressUnlockCondition it should be changed, the others should be as before

File: .gitignore (hunk 16 above)
Comment (Dr-Electron) [reply .gitignore]: I would only add an ignore in the example folder. All other lock files should still be committed, right?

File: bindings/nodejs/examples/how_tos/nfts/send_nft.ts [hunk 19]
```diff
@@ -47,7 +47,7 @@ async function run() {
             },
         ];
 
-        console.log(`Sending NFT '${nftId}' to '${RECV_ADDRESS}'...`);
+        // console.log(`Sending NFT ${nftId} to ${RECV_ADDRESS}...`);
```
Comment (Thoralf-M) [inline bindings/nodejs/examples/how_tos/nfts/send_nft.ts]: Delete if not wanted

File: bindings/nodejs/examples/how_tos/nfts/mint_nft.ts [hunk 20]
```diff
@@ -54,7 +54,7 @@ async function run() {
         // We send from the first address in the account.
         const senderAddress = (await account.addresses())[0].address;
 
-        console.log('Sending the minting transaction for NFT 1...');
+        // console.log('Sending the minting transaction for NFT 1...');
```
Comment (Thoralf-M) [inline bindings/nodejs/examples/how_tos/nfts/mint_nft.ts]: Commented code should be removed if not wanted, same for the two parts below

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Conflicts

**Comment by thibault-martinez** [comment]
Is this still being worked on?

**Comment by anistark** [comment]
> Is this still being worked on?

Yes. Stuck on building Python bindings.

**Comment by thibault-martinez** [comment]
How can we help?

**Comment by Dr-Electron** [comment]
Can we merge it anyway? I mean the example works, right? And building python bindings is something we should discuss in another issue or somewhere else. That way this example isn't blocked as most should run it with the released pypi package anyway

**Comment by thibault-martinez** [comment]
Sure

**Comment by Thoralf-M** [comment]
There are still 2 unresolved comments (and a conflict)

File: sdk/examples/how_tos/nfts/mint_nft.rs [hunk 21]
```diff
@@ -39,15 +33,14 @@ async fn main() -> Result<()> {
...
 
+    // Create the wallet
     let wallet = Wallet::builder()
         .with_storage_path(&var("WALLET_DB_PATH").unwrap())
         .finish()
         .await?;
-    let account = wallet.get_account("Alice").await?;
 
-    // May want to ensure the account is synced before sending a transaction.
-    let balance = account.sync(None).await?;
```
Comment (Thoralf-M) [inline sdk/examples/how_tos/nfts/mint_nft.rs:49]: Bring the sync() back below?

File: sdk/examples/how_tos/nfts/burn_nft.rs [hunk 22]
```diff
@@ -33,14 +29,14 @@ async fn main() -> Result<()> {
...
-        println!("NFTs BEFORE destroying:\n{nfts_before:#?}",);
+        println!("Balance before burning:\n{nfts_before:#?}",);
 
         // Set the stronghold password
         wallet
             .set_stronghold_password(var("STRONGHOLD_PASSWORD").unwrap())
             .await?;
 
-        println!("Sending NFT burn transaction...");
+        // println!("Sending NFT burn transaction...");
```
Comment (Thoralf-M) [inline sdk/examples/how_tos/nfts/burn_nft.rs]: Remove if not needed

**Review by Thoralf-M — CHANGES_REQUESTED** [review]
Please format Rust and nodejs, then I will approve :trollface:

**Review by Thoralf-M — DISMISSED** [review]

File: .gitignore (hunk 16 above)
Comment (thibault-martinez) [reply .gitignore]: Looks like this is back?

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
`yarn.lock` is back in the gitignore

File: .gitignore (hunk 16 above)
Comment (anistark) [reply .gitignore]: removed it.

File: sdk/examples/how_tos/nfts/send_nft.rs [hunk 23]
```diff
@@ -1,15 +1,10 @@
...
+//! Rename `.env.example` to `.env` first.
 //!
-//! Make sure that `example.stronghold` and `example.walletdb` already exist by
-//! running the `create_account` example!
-//!
-//! Rename `.env.example` to `.env` first, then run the command:
-//! ```sh
-//! cargo run --release --all-features --example send_nft
-//! ```
+//! `cargo run --example send_nft --release`
```
Comment (thibault-martinez) [inline sdk/examples/how_tos/nfts/send_nft.rs]:
```suggestion
//! `cargo run --release --example send_nft`
```

File: sdk/examples/how_tos/nfts/mint_nft.rs [hunk 24]
```diff
@@ -1,15 +1,10 @@
...
+//! Rename `.env.example` to `.env` first.
 //!
-//! Make sure that `example.stronghold` and `example.walletdb` already exist by
-//! running the `create_account` example!
-//!
-//! Rename `.env.example` to `.env` first, then run the command:
-//! ```sh
-//! cargo run --release --all-features --example mint_nft
-//! ```
+//! `cargo run --example mint_nft --release`
```
Comment (thibault-martinez) [inline sdk/examples/how_tos/nfts/mint_nft.rs]:
```suggestion
//! `cargo run --release --example mint_nft`
```

**Review by Thoralf-M — DISMISSED** [review]

File: sdk/examples/how_tos/nfts/burn_nft.rs [hunk 25]
```diff
@@ -1,15 +1,10 @@
...
+//! Rename `.env.example` to `.env` first.
 //!
-//! Make sure that `example.stronghold` and `example.walletdb` already exist by
-//! running the `create_account` example!
-//!
-//! Rename `.env.example` to `.env` first, then run the command:
-//! ```sh
-//! cargo run --release --all-features --example burn_nft
-//! ```
+//! `cargo run --example burn_nft --release`
```
Comment (thibault-martinez) [inline sdk/examples/how_tos/nfts/burn_nft.rs]:
```suggestion
//! `cargo run --release --example burn_nft`
```

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**